src/main/java/com/google/api/codegen/transformer/java/JavaSurfaceTestTransformer.java

```diff
@@ -327,17 +356,29 @@ private MockServiceImplFileView createMockServiceImplFileView(InterfaceContext c
         pathMapper.getOutputPath(
             context.getInterfaceModel().getFullName(), context.getProductConfig());
     String name = namer.getMockGrpcServiceImplName(context.getInterfaceModel());
-    String grpcClassName =
-        context
-            .getImportTypeTable()
-            .getAndSaveNicknameFor(namer.getGrpcServiceClassName(context.getInterfaceModel()));
+    String rpcClassName;
+    if (context.getApiModel().getApiSource().equals(ApiSource.PROTO)) {
+      rpcClassName =
+          context
+              .getImportTypeTable()
+              .getAndSaveNicknameFor(
+                  namer.getRpcServiceClassName(
+                      context.getInterfaceModel(), context.getProductConfig()));
+    } else {
+      rpcClassName =
+          context
+              .getImportTypeTable()
+              .getAndSaveNicknameFor(
+                  namer.getRpcServiceClassName(
+                      context.getInterfaceModel(), context.getProductConfig()));
+    }
```

Comment:
These two code branches look the same to me...

Reply:
Merged the two branches!